pkg/signaling/client.go

```diff
@@ -0,0 +1,67 @@
+package signaling
+
+import (
+	"github.com/sirupsen/logrus"
+	"maunium.net/go/mautrix"
+	"maunium.net/go/mautrix/event"
+)
+
+type MatrixClient struct {
+	client *mautrix.Client
+}
+
+func NewMatrixClient(config Config) *MatrixClient {
+	client, err := mautrix.NewClient(config.HomeserverURL, config.UserID, config.AccessToken)
+	if err != nil {
+		logrus.WithError(err).Fatal("Failed to create client")
+	}
+
+	whoami, err := client.Whoami()
+	if err != nil {
+		logrus.WithError(err).Fatal("Failed to identify SFU user")
+	}
+
+	if config.UserID != whoami.UserID {
+		logrus.WithField("user_id", config.UserID).Fatal("Access token is for the wrong user")
+	}
+
+	logrus.WithField("device_id", whoami.DeviceID).Info("Identified SFU as DeviceID")
+	client.DeviceID = whoami.DeviceID
+
+	return &MatrixClient{
+		client: client,
+	}
+}
+
+// Starts the Matrix client and connects to the homeserver,
+// Returns only when the sync with Matrix fails.
+func (m *MatrixClient) RunSync(callback func(*event.Event)) {
```

Comment:
As far as I understand it, this starts syncing in a loop, could the function name indicate that?

Reply:
`RunSync()` makes it sound like it runs sync only once and then stops which is not the case - it starts syncing and keeps doing that until it encounters an error which makes me think `StartSyncing` (or `SyncUntilError`) would be better

Reply:
`RunSyncing` sounds good

Reply:
Ok, renamed!